### PR TITLE
Save isActive in url

### PR DESCRIPTION
**Pull request by bazelajoanna**

### CODE REVIEW

File: app/javascript/react/components/SessionFilters/DormantToggle.tsx [hunk 1]
```diff
@@ -9,55 +8,124 @@ import { useAppDispatch, useAppSelector } from "../../store/hooks";
...
+
   const handleToggleClick = useCallback(() => {
-    const newDormantState = !isDormant;
-    setIsDormant(newDormantState);
-    const currentYear = new Date().getFullYear();
+    const isSettingDormant = !isDormantParameterInUrl;
+    const fixedSessionsTypeToBeSet = isSettingDormant
+      ? FixedSessionsTypes.DORMANT
+      : FixedSessionsTypes.ACTIVE;
+    const urlParamValue = isSettingDormant ? "false" : "true";
```
Comment (evulins) [inline app/javascript/react/components/SessionFilters/DormantToggle.tsx:44]: I would move strings to const:
```
const TRUE = "true";
const FALSE = "false";
```

File: app/javascript/react/components/SessionFilters/ParameterFilter.tsx [hunk 2]
```diff
@@ -167,24 +227,46 @@ interface MobileDeviceParameterFilterProps {
   fetchableSessionsCount: number;
 }
 
-export const MobileDeviceParameterFilter = ({
-  customParameters,
-  sessionsCount,
-  onClose,
-  fetchableSessionsCount,
-}: MobileDeviceParameterFilterProps) => {
+// MobileDeviceParameterFilter Component
```
Comment (evulins) [inline app/javascript/react/components/SessionFilters/ParameterFilter.tsx:230]: Do we need all the comments in this file?

File: app/javascript/react/components/SessionFilters/ParameterFilter.tsx [hunk 3]
```diff
@@ -23,34 +23,90 @@ import {
...
   parameters: string[],
-  sessionType: string
+  sessionType: SessionType
+) => {
+  const basicParameters = getBasicMeasurementTypes(sessionType);
+  return parameters.filter((param) => !basicParameters.includes(param));
+};
+
+// Reusable function to set parameters and URL parameters
+const setParameterParams = (
```
Comment (evulins) [inline app/javascript/react/components/SessionFilters/ParameterFilter.tsx:50]: Can we move this function to mapParams?

File: app/javascript/react/components/SessionFilters/ParameterFilter.tsx [hunk 4]
```diff
@@ -23,34 +23,90 @@ import {
...
+    {
+      key: UrlParamsTypes.currentZoom,
+      value: UrlParamsTypes.previousZoom,
+    },
+  ];
+
+  setUrlParams(commonParams);
 
-  return parameters.filter((param: string) => !basicParameters.includes(param));
+  if (isIndoor === "true" && sensorData.sensorName.startsWith("Gov")) {
```
Comment (evulins) [inline app/javascript/react/components/SessionFilters/ParameterFilter.tsx:94]: move string "true" to const. Maybe we can use `const ` for "Gov" that we already have in file `sensor.ts`

File: app/javascript/react/components/SessionFilters/SensorFilter.tsx [hunk 5]
```diff
@@ -48,31 +49,80 @@ export const getBasicSensors = (
...
-  return sensorsFiltered.map((sensor: Sensor) => sensor.sensorName);
+  return sensorsForMeasurementType
+    .filter((sensor) => !basicSensors?.includes(sensor.sensorName))
+    .map((sensor) => sensor.sensorName);
 };
 
-interface SensorFilterProps {
-  isBasicOpen: boolean;
-}
+const setSensorParams = (
```
Comment (evulins) [inline app/javascript/react/components/SessionFilters/SensorFilter.tsx:71]: Can we extract also this function?

**Review by evulins — COMMENTED** [review]
Good job! 👍  I left some minor comments :)

File: app/javascript/react/components/SessionFilters/SensorFilter.tsx [hunk 6]
```diff
@@ -184,26 +168,30 @@ interface MobileDeviceSensorFilterProps {
...
 
-  const fixedSessionTypeSelected: boolean = sessionType === SessionTypes.FIXED;
-
-  const handleSelectSensor = (selectedSensor: string) => {
-    updateSensorName(selectedSensor, sensors);
-    dispatch(setFetchingData(true));
-  };
+  const handleSelectSensor = useCallback(
+    (selectedSensor: string) => {
+      setSensorParams(selectedSensor, sensors, dispatch);
```
Comment (evulins) [inline app/javascript/react/components/SessionFilters/SensorFilter.tsx:191]: Do we need to pass dispatch?

File: app/javascript/react/components/Map/Map.tsx [hunk 7]
```diff
@@ -369,6 +370,10 @@ const Map = () => {
     }
     !isFirstRender.current && setPreviousZoomOnTheMap();
     isMobile && setPreviousZoomInTheURL();
+
+    setPreviousZoomOnTheMap();
```
Comment (evulins) [inline app/javascript/react/components/Map/Map.tsx:374]: probably we don't need this

File: app/javascript/react/components/SessionFilters/SensorFilter.tsx (hunk 6 above)
Comment (bazelajoanna) [reply app/javascript/react/components/SessionFilters/SensorFilter.tsx:191]: of course, no... thanks!!!

**Review by evulins — APPROVED** [review]
🦃 🥇 Good to go

**Pull request closed without merging**